robottelo/datafactory.py

```diff
@@ -288,3 +289,77 @@ def valid_usernames_list():
         gen_string("alphanumeric"),
         gen_string("numeric"),
     ]
+
+
+def valid_http_credentials(url_encoded=None):
+    """Returns a list of valid credentials for HTTP authentication
+
+    :param url_encoded: flag for quoting special characters
+    :return: A list of dictionaries with user and password credentials
+    """
+    credentials = [
+        {u'login': 'admin', u'pass': 'changeme', u'quote': False},
+        {u'login': '@dmin', u'pass': 'changeme', u'quote': True},
+        {u'login': 'adm/n', u'pass': 'changeme', u'quote': False},
+        {u'login': 'admin2', u'pass': 'ch@ngeme', u'quote': True},
+        {u'login': 'admin3', u'pass': 'chan:eme', u'quote': False},
+        {u'login': 'admin4', u'pass': 'chan/eme', u'quote': True},
+        {u'login': 'admin5', u'pass': 'ch@n:eme', u'quote': True},
+        {u'login': '0', u'pass': 'mypassword', u'quote': False},
+        {u'login': '0123456789012345678901234567890123456789',
+         u'pass': 'changeme',
+         u'quote': False},
+    ]
+    if url_encoded is True:
+        return [{u'login': quote_plus(cred['login'], ''),
+                u'pass': quote_plus(cred['pass'], '')} for cred in credentials
+                ]
+    else:
+        return credentials
+
+
+def incorrect_http_credentials(url_encoded=None):
+    """Returns a list of valid but incorrect credentials for HTTP authentication
```

Comment:
can you help me out? I want to point out to the credentials, which are valid in terms of containing characters (they follow the rules for usernames and passwords) but are incorrect in the meaning of the authentication setup (you can't pass the authentication with them).

As we have 2 scenarios here
- invalid (url can't be even saved with such names/passwords as the url renders invalid)
- incorrect (url can be saved as the url is valid, but the synchronization fails as i is unable to pass authentication)